vista_2d/components.py

```diff
@@ -0,0 +1,352 @@
+
+            if lm <= 255:
+                label = label.astype(np.uint8)
+            elif lm <= 65535:
+                label = label.astype(np.uint16)
+            else:
+                label = label.astype(np.uint32)
+
+            tifffile.imwrite(output_filepath, label)
+            logger.info(f"Saving {output_filepath}")
+
+            polygons = []
+            if overlayed_masks:
+                logger.info(f"Overlay Masks: Reading original Image: {filename}")
+                image = cv2.imread(filename)
+                mask = cv2.imread(output_filepath, 0)
+
+                for i in range(1, np.max(mask)):
+                    m = np.zeros_like(mask)
+                    m[mask == i] = 1
+                    color = np.random.choice(range(256), size=3).tolist()
+                    contours, _ = cv2.findContours(
+                        m, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_NONE
+                    )
+                    polygons.extend(self.to_polygons(contours))
+                    cv2.drawContours(image, contours, -1, color, 1)
+                cv2.imwrite(output_filepath, image)
+                logger.info(f"Overlay Masks: Saving {output_filepath}")
+            else:
+                contours, _ = cv2.findContours(
+                    label, cv2.RETR_LIST, cv2.CHAIN_APPROX_SIMPLE
+                )
+                polygons.extend(self.to_polygons(contours))
+
+            meta_json = {"image_size": image_size, "contours": len(polygons)}
+            with open(os.path.join(output_dir, "meta.json"), "w") as fp:
+                json.dump(meta_json, fp, indent=2)
+
+            if output_contours:
+                logger.info(f"Total Polygons: {len(polygons)}")
+                with open(os.path.join(output_dir, "contours.json"), "w") as fp:
+                    json.dump(
+                        {"count": len(polygons), "contours": polygons}, fp, indent=2
+                    )
+
+        return d
+
+
+# Loss (adopted from Cellpose)
+class CellLoss:
+    def __call__(self, y_pred, y):
+        loss = 0.5 * F.mse_loss(
+            y_pred[:, 1:], 5 * y[:, 1:]
+        ) + F.binary_cross_entropy_with_logits(y_pred[:, [0]], y[:, [0]])
+        return loss
+
+
+# Accuracy (adopted from Cellpose)
+class CellAcc:
+    def __call__(self, mask_pred, mask_true):
+        if isinstance(mask_true, torch.Tensor):
+            mask_true = mask_true.cpu().numpy()
+
+        if isinstance(mask_pred, torch.Tensor):
+            mask_pred = mask_pred.cpu().numpy()
+
+        # print("CellAcc mask_true", mask_true.shape, 'max', np.max(mask_true), ",
+        #       "'mask_pred', mask_pred.shape,  'max', np.max(mask_pred) )
+
+        iou = _intersection_over_union(mask_true, mask_pred)[1:, 1:]
+        tp = _true_positive(iou, th=0.5)
+
+        fp = np.max(mask_pred) - tp
+        fn = np.max(mask_true) - tp
+        ap = tp / (tp + fp + fn)
+
+        # print("CellAcc ap", ap, 'tp', tp, 'fp', fp,  'fn', fn)
+        return ap
```

Comment:
Missing blank line here.

Reply:
Im not sure what you mean, do you want a blank line or do you want me to delete the blank line? @KumoLiu

Reply:
Need a blank line.

Reply:
Added